cg/meta/demultiplex/demux_post_processing.py

```diff
@@ -126,10 +124,116 @@ def finish_flow_cell_temp(self, flow_cell_name: str) -> None:
+        )
 
         # 6. Create sequencing metrics
         self.add_sample_lane_sequencing_metrics_for_flow_cell(flow_cell_name=flow_cell_name)
 
+    def add_flow_cell_data_to_housekeeper(
+        self, flow_cell: Flowcell, flow_cell_directory: Path
+    ) -> None:
+        """Add flow cell data to housekeeper."""
+        LOG.info(f"Add flow cell data to housekeeper for {flow_cell.name}")
+
+        self.add_bundle_and_version_if_not_exists(flow_cell_name=flow_cell.name)
+
+        tags: List[str] = [SequencingFileTag.FASTQ, SequencingFileTag.SAMPLE_SHEET, flow_cell.name]
+        self.ensure_tags_exist(tag_names=tags)
+
+        self.add_sample_sheet(flow_cell_directory, flow_cell.name)
+        self.add_fastq_files(flow_cell_directory=flow_cell_directory, flow_cell_name=flow_cell.name)
+
+    def add_fastq_files(self, flow_cell_directory: Path, flow_cell_name: str) -> None:
+        """Add fastq files from flow cell to housekeeper."""
+        fastq_file_paths: List[Path] = self.get_sample_fastq_file_paths(
+            flow_cell_directory=flow_cell_directory
+        )
+
+        for fastq_file_path in fastq_file_paths:
+            sample_id: str = self.get_sample_id_from_sample_fastq_file_path(
+                fastq_file_path=fastq_file_path
+            )
+
+            if sample_id:
+                self.add_file_if_not_exists(
+                    file_path=fastq_file_path,
+                    flow_cell_name=flow_cell_name,
+                    tag_names=[SequencingFileTag.FASTQ, sample_id],
+                )
+
+    def add_sample_sheet(self, flow_cell_directory: Path, flow_cell_name: str) -> None:
+        """Add sample sheet to housekeeper."""
+        self.add_file_if_not_exists(
+            flow_cell_directory / DemultiplexingDirsAndFiles.SAMPLE_SHEET_FILE_NAME,
+            flow_cell_name,
+            [SequencingFileTag.SAMPLE_SHEET, flow_cell_name],
+        )
+
+    def is_valid_sample_fastq_filename(self, filename: str) -> bool:
+        """Validate the file name and discard any undetermined fastq files."""
+        parts = filename.split("_")
+        return len(parts) > 1 and "Undetermined" not in filename
+
+    def get_sample_fastq_file_paths(self, flow_cell_directory: Path) -> List[Path]:
+        """Get fastq file paths for flow cell."""
+        valid_sample_fastq_file_paths = [
+            file_path
+            for file_path in flow_cell_directory.glob("**/*.fastq.gz")
+            if self.is_valid_sample_fastq_filename(file_path.name)
+        ]
+        return valid_sample_fastq_file_paths
+
+    def get_sample_id_from_sample_fastq_file_path(self, fastq_file_path: Path) -> Optional[str]:
+        """Extract sample id from fastq file name."""
+        sample_directory: str = fastq_file_path.parent.name
+        directory_parts: str = sample_directory.split("_")
+
+        if len(directory_parts) > 1:
+            sample_internal_id = directory_parts[1]
```

Comment:
Is the sample id always the second element in the fastq files parent directory name?

Reply:
Yes.